server/src/internalClusterTest/java/org/elasticsearch/readiness/ReadinessClusterIT.java

```diff
@@ -397,24 +396,24 @@ public void testReadyAfterCorrectFileSettings() throws Exception {
     }
 
     private void causeClusterStateUpdate() {
-        PlainActionFuture.get(
-            fut -> internalCluster().getCurrentMasterNodeInstance(ClusterService.class)
-                .submitUnbatchedStateUpdateTask("poke", new ClusterStateUpdateTask() {
-                    @Override
-                    public ClusterState execute(ClusterState currentState) {
-                        return ClusterState.builder(currentState).build();
-                    }
-
-                    @Override
-                    public void onFailure(Exception e) {
-                        assert false : e;
-                    }
-
-                    @Override
-                    public void clusterStateProcessed(ClusterState initialState, ClusterState newState) {
-                        fut.onResponse(null);
-                    }
-                })
-        );
+        final var latch = new CountDownLatch(1);
+        internalCluster().getCurrentMasterNodeInstance(ClusterService.class)
+            .submitUnbatchedStateUpdateTask("poke", new ClusterStateUpdateTask() {
+                @Override
+                public ClusterState execute(ClusterState currentState) {
+                    return ClusterState.builder(currentState).build();
+                }
+
+                @Override
+                public void onFailure(Exception e) {
+                    assert false : e;
+                }
+
+                @Override
+                public void clusterStateProcessed(ClusterState initialState, ClusterState newState) {
+                    latch.countDown();
+                }
+            });
+        safeAwait(latch);
```

Comment:
Is there a reason you didn't use `safeAwait(Consumer<ActionListener>)` here? I've noticed that in a few places and on the surface appears a bit simpler.

Reply:
It's a bit less nesting, and indicates that we're not capturing any failures here. Don't really have a strong preference tho.